src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -193,6 +193,17 @@ class DefaultStorageKeyAllocator
         return StorageKeyName::Formatted("f/%x/icd/%x", fabric, index);
     }
 
+    // Thread Network Directory
+
+    static StorageKeyName ThreadNetworkDirectoryIndex() { return StorageKeyName::FromConst("g/tnd/i"); }
+    static StorageKeyName ThreadNetworkDirectoryDataset(uint64_t extendedPanId)
+    {
+        return StorageKeyName::Formatted("g/tnd/n/%08" PRIx32 "%08" PRIx32, // some platforms can't format uint64
```

Comment:
Please update `IsMemoryOnlyGlobalKey` in the Darwin framework to document why those keys will not be observed there?

Reply:
Done